### PR TITLE
Add support for posting/unposting links

### PR DESCRIPTION
**Pull request by nickeidler**

### CODE REVIEW

File: lib/rentlinx/modules/linkable.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      post
+      post_links
+    end
+
+    def post_links
+      Rentlinx.client.post_links(@links)
+    end
+
+    def links
+      return @links if @links
```
Comment (d-ray) [inline lib/rentlinx/modules/linkable.rb]: You could remove this line by using <code>@links ||= if defined? unitID …</code>

File: test/modules/test_link_client_methods.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+      new_links = Rentlinx.client.get_links_for_unit(@unit)
+      assert_equal 1, new_links.size
+    end
+  end
+
+  def test_post_invalid_link_raises_an_exception
+    use_vcr do
+      invalid_link = Rentlinx::UnitLink.new(unitID: @unit.unitID, url: 'I am not a URL!')
+      assert_raises(Rentlinx::InvalidObject) do
```
Comment (d-ray) [inline test/modules/test_link_client_methods.rb]: Can we also check the error message to make sure the object is invalid for the expected reason?

**Comment by d-ray** [comment]
:+1:

File: lib/rentlinx/modules/link_client_methods.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          UnitLink.new(symbolize_data(link_data))
+        end
+      end
+    rescue Rentlinx::NotFound
+      return []
+    end
+
+    def get_links_for_unit(unit)
+      links = get_links_for_property_id(unit.propertyID)
+      links.select { |a| defined? a.unitID && a.unitID == unit.unitID }
```
Comment (d-ray) [inline lib/rentlinx/modules/link_client_methods.rb]: Does the letter a have any significance?  In this situation, it's easy to tell what the variable is, but I still prefer names that are at least slightly meaningful.

File: lib/rentlinx/modules/linkable.rb (hunk 1 above)
Comment (scuendet) [inline lib/rentlinx/modules/linkable.rb]: I think the current way is clearer though.

Comment (d-ray) [inline lib/rentlinx/modules/linkable.rb]: You could make that argument.  Using <code>||=</code> is a very common ruby pattern for this kind of thing though:  http://www.justinweiss.com/blog/2014/07/28/4-simple-memoization-patterns-in-ruby-and-one-gem/.  It doesn't matter that much to me, though.

**Pull request merged**